internal/wazeroir/operations.go

```diff
@@ -948,19 +924,39 @@ func (b BranchTargetDrop) String() (ret string) {
 	return
 }
 
-// OperationUnreachable implements Operation.
+// UnionOperation implements Operation and is the compilation (engine.lowerIR) result of a wazeroir.Operation.
 //
-// This corresponds to wasm.OpcodeUnreachable.
+// Not all operations result in a UnionOperation, e.g. wazeroir.OperationI32ReinterpretFromF32, and some operations are
+// more complex than others, e.g. wazeroir.OperationBrTable.
 //
-// The engines are expected to exit the execution with wasmruntime.ErrRuntimeUnreachable error.
-type OperationUnreachable struct{}
+// Note: This is a form of union type as it can store fields needed for any operation. Hence, most fields are opaque and
+// only relevant when in context of its kind.
+type UnionOperation struct {
+	// OpKind determines how to interpret the other fields in this struct.
+	OpKind   OperationKind
```

Comment:
we can't, because there is already `Kind()`. In order to make migration smooth I am letting `UnionOperation` implement `Operation`. When no other `Operation` will be left, we can remove the interface and use the field directly, losing the need for the `Op` prefix

Reply:
oh ic make sense!